frontend/packages/dev-console/src/components/add/AddCardItem.tsx

```diff
@@ -0,0 +1,64 @@
+import * as React from 'react';
+import { useUserSettings } from '@console/shared';
+import { SimpleListItem, Title, Text } from '@patternfly/react-core';
+import { ResolvedExtension, AddAction } from '@console/dynamic-plugin-sdk';
+import { navigateTo, resolvedHref } from '../../utils/add-page-utils';
+import './AddCardItem.scss';
+
+type AddCardItemProps = {
+  action: ResolvedExtension<AddAction>;
+  namespace: string;
+};
+
+const AddCardItem: React.FC<AddCardItemProps> = ({
+  action: {
+    properties: { id, label, icon, href, description },
+  },
+  namespace,
+}) => {
+  const [showDetails] = useUserSettings('console.addPage.showDetails', 'show', true);
```

Comment:
`useUserSettings` didn't exhibit expected behavior while using boolean. Although the value was being stored in storage, but the component didn't re-render resulting in no change in the UI.
cc @jerolimov

Reply:
The hook was also used at another spot, so it make sense that you extract this into an own hook:

```tsx
export const useShowDetails = (): [boolean, (value: boolean) => void] => {
  const [showDetails, setShowDetails] = useUserSettings(
    'console.addPage.showDetails',
    'show',
    true,
  );
  return [showDetails === 'show', (value: boolean) => setShowDetails(value ? 'show' : 'hide')];
};
```

As we discussed earlier, I now researched the issue that a boolean could not be saved in user settings. It looks like this happen only in the localStorage variant. You can test the "real" user settings (which are stored on the cluster in a ConfigMap) when running the bridge with this parameter. I will create an issue and PR for that so that we can review this independently.

```bash
bin/bridge -user-settings-location configmap
```